src/shim/gas.rs

```diff
@@ -113,36 +128,62 @@ impl From<GasChargeV2> for GasCharge {
 
 impl From<GasChargeV3> for GasCharge {
     fn from(value: GasChargeV3) -> Self {
+        GasChargeV4 {
+            name: value.name,
+            compute_gas: GasV4::from_milligas(value.compute_gas.as_milligas()),
+            other_gas: GasV4::from_milligas(value.other_gas.as_milligas()),
+            elapsed: value.elapsed.get().map(|&d| d.into()).unwrap_or_default(),
+        }
+        .into()
+    }
+}
+
+impl From<GasChargeV4> for GasCharge {
+    fn from(value: GasChargeV4) -> Self {
         GasCharge(value)
     }
 }
 
-impl From<GasCharge> for GasChargeV3 {
+impl From<GasCharge> for GasChargeV2 {
     fn from(value: GasCharge) -> Self {
-        value.0
+        Self {
+            name: value.0.name,
+            compute_gas: GasV2::from_milligas(value.0.compute_gas.as_milligas() as _),
+            storage_gas: GasV2::from_milligas(value.0.other_gas.as_milligas() as _),
+        }
     }
 }
 
-impl From<GasCharge> for GasChargeV2 {
+impl From<GasCharge> for GasChargeV3 {
     fn from(value: GasCharge) -> Self {
-        GasChargeV2 {
+        Self {
             name: value.0.name,
-            compute_gas: GasV2::from_milligas(value.0.compute_gas.as_milligas() as i64),
-            storage_gas: GasV2::from_milligas(value.0.other_gas.as_milligas() as i64),
+            compute_gas: GasV3::from_milligas(value.0.compute_gas.as_milligas() as _),
+            other_gas: GasV3::from_milligas(value.0.other_gas.as_milligas() as _),
+            // FIXME: https://github.com/ChainSafe/forest/issues/3524
+            elapsed: Default::default(),
```

Comment:
can we fix it now?

Reply:
Unfortunately `GasDuration` is not exported from fvm@3, I'm thinking about creating a macro that implements `From` with unsafe `std::mem` API to simplify type conversions between fvm@3 and fvm@4 given most of the type pairs are identical between them